lib/views/post/widgets/post_buttons.dart

```diff
@@ -57,114 +125,170 @@ class _PostButtonsState extends State<PostButtons> {
   void initState() {
     _voteService.getVoteOfItem(widget.info.item).then((vote) {
       setState(() {
+        if (vote == Vote.up || vote == Vote.favorite) {
+          _upAnimationName = "voted";
+          if(vote == Vote.up) {
+            _favoriteAnimationName = "unfocus";
+            _downAnimationName = "unfocus";
+          }
+        }
+        if (vote == Vote.down) {
+          _downAnimationName = "voted";
+          _favoriteAnimationName = "unfocus";
+          _upAnimationName = "unfocus";
+        }
+        if (vote == Vote.favorite) {
+          _favoriteAnimationName = "voted";
+          _upAnimationName = "unfocus";
+          _downAnimationName = "unfocus";
+        }
         currentVote = vote;
+        init = false;
       });
     });
     super.initState();
   }
 
+  bool init = true;
+
   @override
   Widget build(BuildContext context) {
     final loggedIn = GlobalInherited.of(context).isLoggedIn;
-    final votedColor = new Color(0xffee4d2e);
-
-    return Row(
-      children: [
-        IconButton(
-          icon: Icon(Icons.add_circle_outline),
-          color: currentVote == Vote.up || currentVote == Vote.favorite
-              ? votedColor
-              : Colors.white,
-          onPressed: loggedIn ? () => voteItem(Vote.up) : null,
-          disabledColor: Colors.white30,
-        ),
-        IconButton(
-          color: currentVote == Vote.down ? votedColor : Colors.white,
-          icon: Icon(Icons.remove_circle_outline),
-          onPressed: loggedIn ? () => voteItem(Vote.down) : null,
-          disabledColor: Colors.white30,
-        ),
-        IconButton(
-          color: currentVote == Vote.favorite ? votedColor : Colors.white,
-          icon: Icon(
-            currentVote == Vote.favorite
-                ? Icons.favorite
-                : Icons.favorite_border,
+    if (!loggedIn) {
+      _upAnimationName =
+          _downAnimationName = _favoriteAnimationName = "disabled";
+    }
+    if (init) return Center(child: CircularProgressIndicator());
```

Comment:
Why do you need the init? initState should be called before the build method.

Reply:
yes called before build, but it is not completed when build is beeing called.

Reply:
@mathis-m probably due to the "setState", which is redundant in the initState method as far as I know. Try it without the setState and see if it works without the "init" guard variable.

Reply:
I introduced it because the flare animation was in a bug state. Maybe can be removed since i changed the flare files